troupon/static/js/location.js

```diff
@@ -0,0 +1,75 @@
+    $(document).ready(function() {
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(showPosition, showError);
+      } else {
+        toastr.error("Geolocation is not supported by this browser.");
+      }
+
+      function showError(error) {
+        switch (error.code) {
+          case error.PERMISSION_DENIED:
+            toastr.error("You have denied Troupon access to your location")
+            deleteCookie();
+            break;
+          case error.POSITION_UNAVAILABLE:
+            toastr.error("Location information is unavailable.")
+            deleteCookie();
+            break;
+          case error.TIMEOUT:
+            toastr.error("The request to get user location timed out.")
+            deleteCookie();
+            break;
+          case error.UNKNOWN_ERROR:
+            toastr.error("An unknown error occurred.")
+            deleteCookie();
+            break;
+        }
```

Comment:
You could reduce duplications here by calling the `toaster.error(errorMessage)` and the `deleteCookie()` just once at the end of the `showError` function.

You could simply use the switch block to decide what error message to display.

For example:

```
function showError(error) {
  var errorMsg = '';
  switch(error.code) {
    case error.PERMISSION_DENIED:
      errorMsg = "You have denied Troupon access to your location";
      break;
    case ...
    ...
  }
  toastr.error(errorMsg);
  deleteCookie();
}
```